damn-fast-updatable-priority-queue/test.lisp

```diff
@@ -0,0 +1,246 @@
+;;;; damn-fast-updatable-priority-queue-test.lisp
+
+(defpackage #:damn-fast-updatable-priority-queue/test
+  (:use #:cl)
+  (:local-nicknames (#:a #:alexandria)
+                    (#:q #:damn-fast-updatable-priority-queue))
+  (:export #:run))
+
+(in-package #:damn-fast-updatable-priority-queue/test)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;; Utilities
+
+(defun verify-heap-property (vector)
+  (declare (type q::prio-vector-type vector))
+  (loop with length = (length vector)
+        for parent from 0 below (truncate length 2)
+        for left = (+ (* parent 2) 1)
+        for right = (+ (* parent 2) 2)
+        do (assert (<= (aref vector parent) (aref vector left)) ()
+                   "VERIFY-HEAP-PROPERTY: Invalid left child: ~D -> ~D"
+                   (aref vector parent) (aref vector left))
+        when (oddp length)
+          do (assert (<= (aref vector parent) (aref vector right)) ()
+                     "VERIFY-HEAP-PROPERTY: Invalid right child: ~D -> ~D"
+                     (aref vector parent) (aref vector right))))
+
+(defun verify-heap-property-for-queue (queue)
+  (verify-heap-property (subseq (q::%prio-vector queue)
+                                0 (q:size queue)))
+  (loop with data = (q::%data-vector queue)
+        for i below (q:size queue)
+        for d = (aref data i)
+        do (assert (eq d (aref data (car d))))))
+
+(defun stringify (i) (format nil "~D" i))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;; Main interface
+
+(defvar *verbose* nil)
+
+(defun run (&optional *verbose*)
+  (dolist (length (nconc (a:iota 64 :start 1) '(256 1024 4096)))
+    (when *verbose* (format t "~&Testing with ~4,' D elements" length))
+    (let ((queue (q:make-queue (max 1 (ash length -4)))))
+      (perform-test queue (a:iota length))
+      (perform-test queue (nreverse (a:iota length)))
+      (dotimes (i 100)
+        (perform-test queue (a:shuffle (a:iota length))))))
+  (perform-error-test)
```

Comment:
We seem to be missing the shuffle test here, the one that was added in standard DFPQ tests.